ludwig/api.py

```diff
@@ -104,6 +106,86 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclass
+class EvaluationFrequency:
+    """Represents the frequency of periodic evaluation of a metric during training. For example:
+
+    "every epoch"
+    frequency: 1, period: EPOCH
+
+    "every 50 steps".
+    frequency: 50, period: STEP
+    """
+
+    frequency: float = 1.0
+    period: str = "epoch"  # One of "epoch" or "step".
+
+    EPOCH: ClassVar[str] = "epoch"  # One epoch is a single pass through the training set.
+    STEP: ClassVar[str] = "step"  # One step is training on one mini-batch.
+
+
+@dataclass
+class TrainingStats:
+    """Training stats were previously represented as a tuple or a dict.
+
+    This class replaces those while preserving dict and tuple-like behavior (unpacking, [] access).
+    """
+
+    training: Dict[str, Any]
+    validation: Dict[str, Any]
+    test: Dict[str, Any]
+    evaluation_frequency: EvaluationFrequency = dataclasses.field(default_factory=EvaluationFrequency)
+
+    def __iter__(self):
+        return iter((self.training, self.test, self.validation))
+
+    def __contains__(self, key):
+        return (
+            (key == TRAINING and self.training)
+            or (key == VALIDATION and self.validation)
+            or (key == TEST and self.test)
+        )
+
+    def __getitem__(self, key):
+        # Supports dict-style [] element access for compatibility.
+        return {TRAINING: self.training, VALIDATION: self.validation, TEST: self.test}[key]
```

Comment:
Should we add a Todo here to get rid of these auxiliary/non-standard accessor methods?